packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -289,6 +300,45 @@ export class CfnResource extends CfnRefElement {
     addDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
   }
 
+  /**
+   * Indicates that this resource no longer depends on another resource.
+   *
+   * This can be used for resources across stacks (including nested stacks)
+   * and the dependency will automatically be removed from the relevant scope.
+   */
+  public removeDependency(target: CfnResource) : void {
+    // skip this dependency if the target is not part of the output
+    if (!target.shouldSynthesize()) {
+      return;
+    }
+
+    removeDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
+  }
+
+  /**
+   * Retrieves an array of resources this resource depends on.
+   *
+   * This assembles dependencies on resources across stacks (including nested stacks)
+   * automatically.
+   */
+  public obtainDependencies() {
```

Comment:
Strictly speaking, yes, we could make due with only add/remove methods, but I've found that, especially when developing construct libraries for my developers to leverage, the lack of visibility into certain items in the cdk api, like construct dependencies, makes for a lot of frustration when trying to develop something with more nuance to it, because this is effectively a write-only interface, where there is zero ability to inspect what is actually there, only to attempt to change it.
Without this `obtain` method, a user is forced to blindly apply a `removeDependency` to every node in an entire construct tree, if they aren't sure where a dependency might be that they need to remove. While that approach would probably serve most use cases in the strictest sense, I can imagine that sometimes more nuance could be desired.
With that in mind, I would ask the converse: What is the harm in allowing a user to actually view what they are touching? All told, the `obtain` logic added is really quite limited.